packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -221,7 +221,11 @@ class AnsiTerminal {
   String clearScreen() => supportsColor ? _clear : '\n\n';
 
   set singleCharMode(bool value) {
-    stdin.lineMode = !value;
+    try {
+      stdin.lineMode = !value;
+    } on IOException {
+      // This can throw for some terminals; we ignore the error.
```

Comment:
Yeah, we can extract the `osError.errorCode` from the exception.  The problem is that StdinException wasn't introduced until sdk 1.22, so some transient ugliness is required -- https://github.com/flutter/flutter/pull/7472